steering_vectors/record_activations.py

```diff
@@ -49,22 +64,24 @@ def record_activations(
     matcher = layer_config[layer_type]
     matching_layers = collect_matching_layers(model, matcher)
     hooks: list[RemovableHandle] = []
+    module_names = []
     for layer_num, layer_name in enumerate(matching_layers):
         if layer_nums is not None and layer_num not in layer_nums:
             continue
         module = get_module(model, layer_name)
-        hook_fn = _create_read_hook(
+        hook_fn = _create_read_hook_output(
             layer_num, recorded_activations, clone_activations=clone_activations
         )
         hooks.append(module.register_forward_hook(hook_fn))
+        module_names.append(layer_name + ".output")
     try:
-        yield recorded_activations
+        handle = RecordHandle(recorded_activations, hooks, module_names)
```

Comment:
This looks like a breaking change to a documented, public-facing method. Why is this change needed? It's OK to make breaking changes, but we should bump the major version of the library if we do this, and it should only be done if there's a clear use-case for it.